decide/src/main/java/org/example/Point.java

```diff
@@ -16,4 +16,45 @@ public double distance(Point other) {
     public static double triangleArea(Point p1, Point p2, Point p3) {
         return Math.abs(p1.x * (p2.y - p3.y) + p2.x * (p3.y - p1.y) + p3.x * (p1.y - p2.y)) / 2.0;
     }
+
+    /**
+     * Computes the angle from which Point p is the vertex.
+     * @param p2 first outer point of the angle
+     * @param p3 second outer point of the angle
+     * @return the angle formed by (p,p2) and (p,p3) in rad
+     */
+    public double angle(Point p2, Point p3) {
+        double scalarProduct = (p2.x - x) * (p3.x - x) + (p2.y - y) * (p3.y - y);
+        return Math.acos(scalarProduct / (distance(p2) * distance(p3)));
+    }
+
+    /**
+     * Computes the radius of the smallest circle containing all three points (p1, p2, p3)
+     * If one angle of the triangle is obtuse:
+     * the smallest circle containing the triangle has the opposite side as diameter
+     * Else:
+     * the smallest circle is the circumscribed circle.
+     * the radius is computed thanks to the law of sines : R = a * b * c / (4 * S)
+     * @param p1 first point
+     * @param p2 second point
+     * @param p3 third point
+     * @return radius of the smallest circle containing p1, p2 and p3
+     */
+    public static double smallestCircleRadius(Point p1, Point p2, Point p3) {
+        // checking if any of the angles is obtuse
+        if (p1.angle(p2, p3) > Math.PI) {
```

Comment:
If you want to check if the angle is obtuse shouldn't you check `> Math.PI / 2`?

Reply:
Right, thanks!